### PR TITLE
GH-2054: As a smith I want an infrastructure to run Xpect tests as AbstractIdeTests

### LINKED ISSUE #2054: As a smith I want an infrastructure to run Xpect tests as AbstractIdeTests
Check whether this can be done by providing Xpect preamble constructs to create ide tests.

### PR DESCRIPTION
**Pull request by mmews-n4**
closes #2054

The new JUnit runner does not rely on Xpect anymore. However, it complies to some conventions of the Eclipse Xpect plugin and hence some ui features are still available such as
* navigation from JUnit View to Xpect method
* navigation from Xpect method to implementing method
* starting .xt files to run / debug as JUnit test

To make use of these features, we still use the keyword "XPECT" and the setup section "XPECT_SETUP". Note that this is not required and might be changed in the future (but would result in losing support for the ui features mentioned above).

To support navigation from JUnit View to Xpect method, we conform to the test name scheme required by the Eclipse Xpect plugin, implemented here: https://github.com/eclipse/Xpect/blob/0e5186a5fd96d4c82536fcff9acfa6ffada9fff8/org.eclipse.xpect.ui.junit/src/org/eclipse/xpect/ui/junit/TestDataUIUtil.java#L82

### CODE REVIEW

**Review by mor-n4 — APPROVED** [review]
You started out to repair the roof and built an entire new house! 😃 
Great, this will be very useful in the future!

File: plugins/org.eclipse.n4js.xpect/src/org/eclipse/n4js/xpect/methods/TypeXpectMethod.java [hunk 1]
```diff
@@ -44,7 +27,7 @@
 @XpectImport(N4JSOffsetAdapter.class)
 public class TypeXpectMethod {
 	@Inject
-	private N4JSTypeSystem ts;
+	private XtMethods mh;
```
[the remark this answers is not in the document]
Comment (mmews-n4) [reply plugins/org.eclipse.n4js.xpect/src/org/eclipse/n4js/xpect/methods/TypeXpectMethod.java:30]: As discussed: Until we really switch from using Xpect to using our own implementation (done in this PR) we still need to call those methods.

File: plugins/org.eclipse.n4js.xpect/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -19,7 +19,8 @@ Require-Bundle: org.eclipse.n4js,
  org.eclipse.n4js.utils,
  org.eclipse.n4js.runner,
  org.eclipse.n4js.runner.nodejs,
- org.eclipse.n4js.flowgraphs
+ org.eclipse.n4js.flowgraphs,
+ org.eclipse.n4js.ide.tests.helper
```
[the remark this answers is not in the document]
Comment (mmews-n4) [reply plugins/org.eclipse.n4js.xpect/META-INF/MANIFEST.MF:23]: Thanks for noting this. I'll check and fix that.

**Comment by mmews-n4** [comment]
Thanks for the review! Yes, thanks, you're right - I didn't expect that to happen! ;-)

**Pull request merged**